string.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+type TokenType int
+
+const (
+	TokenLiteral = iota
+	TokenReference
+)
+
+type Token struct {
+	Type  TokenType
+	Value string
+}
+
+// StringReference is an implementation of PostProcessor to de-reference (( grab me.data )) calls
+type StringReferencer struct {
+	root map[interface{}]interface{}
+}
+
+// Action returns the Action string for the Dereferencer
+func (s StringReferencer) Action() string {
+	return "string-referencer"
+}
+
+// PostProcess - resolves a value by seeing if it matches (( grab me.data )) and retrieves me.data's value
+func (s StringReferencer) PostProcess(o interface{}, node string) (interface{}, string, error) {
+	if o != nil && reflect.TypeOf(o).Kind() == reflect.String {
+		re := regexp.MustCompile(`^\Q((\E\s*string\s+(.+)\s*\Q))\E$`)
+		if re.MatchString(o.(string)) {
+			keys := re.FindStringSubmatch(o.(string))
+
+			tokens, err := parseWords(keys[1])
+			if err != nil {
+				return nil, "error", err
```

Comment:
I think this error message might need to be marked up a little more (node name, and reason) - unless parseWords stops returning errors, see below